R/get_dist_parms.R

```diff
@@ -0,0 +1,80 @@
+#' Get distribution parameters
+#' @description Internal helper function for
+#' calculating parametric distribution parameters.
+#' @param data Numeric vector
+#' @param distribution_type Character.
+#' One of either "normal", "lognormal", or "beta"
+#' @note This function will first try fitting using mme, then mle
+#' @importFrom fitdistrplus fitdist gofstat
+#' @keywords internal
+
+get_dist_parms <- function(data, distribution_type) {
+
+  if (length(distribution_type) > 1) {
+    stop("This function only handles on distribution type at a time.")
+
+  }
+
+
+  if (distribution_type == "normal") {
+    try(fit <- fitdist(data = data, distr = "norm",
+                        keepdata = F, method = "mme"))
+    if (!exists("fit")) {
```

Comment:
is it better to use try ... exits, or tryCatch?

Reply:
Good point, this seems like a good use of tryCatch.  I've made the change

Reply:
I've also added a helper function to suppress prevent the unnecessary text that prints when fitdist has trouble fitting a model.  This typically happens when using method = "mle".   Method = "mme" is more robust, but does not return standard errors, so I've kept mle as the default.